static/src/javascripts/projects/common/modules/navigation/newSticky.js

```diff
@@ -53,88 +46,122 @@ define([
                         : getClientAdHeight();
                 };
 
+                var topAdRenderedPromise = new Promise(function (resolve) {
+                    mediator.on('modules:commercial:dfp:rendered', function (event) {
+                        if (event.slot.getSlotElementId() === adId) {
+                            resolve();
+                        }
+                    });
+                });
+
+                var newRubiconAdHeightPromise = new Promise(function (resolve) {
+                    window.addEventListener('message', function (event) {
+                        var data = JSON.parse(event.data);
+                        var $iframe = getAdIframe();
+                        var isRubiconAdEvent = data.type === 'set-ad-height';
```

Comment:
/cc @jbreckmckye @Calanthe @uplne FYI, we've added a `postMessage` to the Rubicon expand script so we can know when an ad changes its size. This might be useful for other things, too :smile: